libgnucash/backend/sql/gnc-sql-column-table-entry.cpp

```diff
@@ -367,151 +367,87 @@ GncSqlColumnTableEntryImpl<CT_GUID>::add_to_query(QofIdTypeConst obj_name,
     }
 }
 /* ----------------------------------------------------------------- */
-typedef Timespec (*TimespecAccessFunc) (const gpointer);
-typedef void (*TimespecSetterFunc) (const gpointer, Timespec*);
-
-#define TIMESPEC_COL_SIZE (4+3+3+3+3+3)
+typedef time64 (*Time64AccessFunc) (const gpointer);
+typedef void (*Time64SetterFunc) (const gpointer, time64);
+constexpr int TIME_COL_SIZE = 4 + 3 + 3 + 3 + 3 + 3;
 
 template<> void
-GncSqlColumnTableEntryImpl<CT_TIMESPEC>::load (const GncSqlBackend* sql_be,
-                                               GncSqlRow& row,
-                                               QofIdTypeConst obj_name,
-                                               gpointer pObject) const noexcept
+GncSqlColumnTableEntryImpl<CT_TIME>::load (const GncSqlBackend* sql_be,
+                                            GncSqlRow& row,
+                                            QofIdTypeConst obj_name,
+                                            gpointer pObject)
+    const noexcept
 {
-
-    Timespec ts = {0, 0};
-    gboolean isOK = FALSE;
-
-
-    g_return_if_fail (pObject != NULL);
+    time64 t{0};
     g_return_if_fail (m_gobj_param_name != nullptr || get_setter(obj_name) != nullptr);
-
     try
     {
-        auto val = row.get_time64_at_col(m_col_name);
-        timespecFromTime64 (&ts, val);
+        t = row.get_time64_at_col (m_col_name);
     }
     catch (std::invalid_argument&)
     {
         try
         {
             auto val = row.get_string_at_col(m_col_name);
             GncDateTime time(val);
-            ts.tv_sec = static_cast<time64>(time);
+            t = static_cast<time64>(time);
         }
-        catch (const std::invalid_argument& err)
+        catch (std::invalid_argument& err)
```

Comment:
Why did you remove the const qualifier ? I have been using it in my catch expressions because I saw plenty of examples using one. But honestly I haven't thought this through so I don't know which one is better or why.

Reply:
I didn't exactly mean to. What I actually did was to delete the CT_TIMESPEC load and change the CT_TIME64 load, and I had to add back in the alternate string handler and missed making it const.